app/services/IntersectService.ts

```diff
@@ -0,0 +1,35 @@
+import { checkIntersect as intersect } from '../helpers/Intersect';
+import { HealthcareAuthority } from '../store/types';
+import BackgroundTaskService from './BackgroundTaskService';
+
+class IntersectService {
+  isServiceRunning = false;
+  nextJob: HealthcareAuthority[] | null = null;
+
+  checkIntersect = (
+    healthcareAuthorities: HealthcareAuthority[] | null,
+    bypassTimer: boolean,
+  ): string => {
+    if (this.isServiceRunning) {
+      this.nextJob = healthcareAuthorities;
+      return 'skipped';
+    }
+    this.isServiceRunning = true;
+
+    intersect(healthcareAuthorities, bypassTimer).then(() => {
+      this.isServiceRunning = false;
+      if (this.nextJob) {
+        const job = this.nextJob;
+        this.nextJob = null;
+        this.checkIntersect(job, bypassTimer);
+      }
+    });
+
+    BackgroundTaskService.start();
```

Comment:
Why do we need to call start every time? Doesn't this include config setting?

Reply:
This will be better addressed in a followup PR.